### DISCUSSION TITLE
Not an issue: Trying to run the FLAIR TensorFlow sample in Google Colab, and having trouble

### DISCUSSION THREAD

**Discussion by hti-kumar** [NVIDIA/NVFlare#225] [Q&A] [answered]
Hello, I am getting started on FLARE. As a first experiment, I was trying to run the Tensor Flow example provided in Google Colab.  I  created the server and client processes and it looks they are all waiting and not running.  (Is there a way to attach Jupyter notebooks here? I would be able to show the code in a more useful form). I believe it is a simple fix -- I need it to go forward. Thanks -Kumar, HTI Inc, Portland Oregon.

[Copy of FLARE TF sample.ipynb - Colaboratory.pdf](https://github.com/NVIDIA/NVFlare/files/8113710/Copy.of.FLARE.TF.sample.ipynb.-.Colaboratory.pdf)

**Comment by yanchengnv** [comment]
Once started, the server and clients will be waiting for your commands. You will need to start the Admin client to start your app.  The instructions for these are in the PDF file you shared, in page 9:

!./poc/admin/startup/fl_admin.sh localhost

Please follow these instructions. Let us know if you run into any issues.

**Reply by hti-kumar** [reply]
I think I did! And verified it is running: 

!./poc/admin/startup/fl_admin.sh localhost
/content/poc/admin/startup
Admin Server: localhost on port 8003
User Name: admin
Password:
Type ? to list commands; type "? cmdName" to show usage of a command.
> check_status server
FL_app name: ?
Engine status: stopped
Run number has not been set.
Registered clients: 2
----------------------------------------------------------------------------
| CLIENT | TOKEN | LAST CONNECT TIME |
----------------------------------------------------------------------------
| site-1 | d2a8e53a-78a9-4e48-b8d1-83dcfc39d8be | Sun Feb 13 18:38:43 2022 |
| site-2 | 9094ec01-1dad-4fdf-92b4-7f922de61b50 | Sun Feb 13 18:38:46 2022 |
----------------------------------------------------------------------------
Done [1532 usecs] 2022-02-13 18:39:38.174908
> upload_app hello-tf2
Created folder /content/poc/server/startup/../transfer/hello-tf2


Is there any other command to be issued to get the clients and server moving?
Regards
--Kumar

**Comment by yanchengnv** [comment, marked as answer]
Yes, there are many more commands. If you type "?" at the prompt you will see them.

Looks like you only started the admin client, but didn't issue commands to upload/deploy/start the app. 

Basically, once the server and FL clients are running, the system can do many experiments (RUNs) - one at a time - controlled by your commands thru the Admin Client: 

- upload the app you want to train
- set a run number for the training. 
- deploy the app
- start the app. 

Once the app is started, you could let it run to completion or stop it (using the abort command) in the middle, and then set another run number to start another app.

Please follow through these steps as shown in the doc.

> upload_app hello-tf2
Created folder /content/poc/server/startup/../transfer/hello-tf2
> set_run_number 1
Created a new run folder: run_1
---------------------
| CLIENT | RESPONSE |
---------------------
| site-1 | OK |
| site-2 | OK |
---------------------
Done [502768 usecs] 2022-02-13 18:43:29.389927
> deploy_app hello-tf2 all
deployed app "hello-tf2" to Server
---------------------
| CLIENT | RESPONSE |
---------------------
| site-1 | OK |
| site-2 | OK |
---------------------
Done [406576 usecs] 2022-02-13 18:43:48.774886
> start_app all
Server app is starting....
---------------------
| CLIENT | RESPONSE |
---------------------
| site-1 | |
| site-2 | |
---------------------
Done [15383932 usecs] 2022-02-13 18:44:22.332943
> check_status server

**Reply by YuanTingHsieh** [reply]
just a note these contents are also available here: https://nvidia.github.io/NVFlare/examples/hello_tf2.html
